internal/fs/fs.go

```diff
@@ -679,31 +743,9 @@ func (fs *fileSystem) mintInode(ic inode.Core) (in inode.Inode) {
 
 	// Create the inode.
 	switch {
-	// Explicit directories
-	case ic.MinObject != nil && ic.FullName.IsDir():
-		in = inode.NewExplicitDirInode(
-			id,
-			ic.FullName,
-			ic.MinObject,
-			fuseops.InodeAttributes{
-				Uid:  fs.uid,
-				Gid:  fs.gid,
-				Mode: fs.dirMode,
-
-				// We guarantee only that directory times be "reasonable".
-				Atime: fs.mtimeClock.Now(),
-				Ctime: fs.mtimeClock.Now(),
-				Mtime: fs.mtimeClock.Now(),
-			},
-			fs.implicitDirs,
-			fs.mountConfig.ListConfig.EnableEmptyManagedFolders,
-			fs.enableNonexistentTypeCache,
-			fs.dirTypeCacheTTL,
-			ic.Bucket,
-			fs.mtimeClock,
-			fs.cacheClock,
-			fs.mountConfig.MetadataCacheConfig.TypeCacheMaxSizeMB,
-			fs.mountConfig.EnableHNS)
+	// Explicit directories or Folders in Hierarchical bucket.
+	case (ic.MinObject != nil && ic.FullName.IsDir()) || ic.Folder != nil:
```

Comment:
wont adding two seperate cases work here:
case ic.MinObject != nil && ic.Full.IsDir():
case ic.Folder != nil:
     in = fs.createExplicitDirInode(id, ic)

Reply:
Didn't you mean the same which made changes for
https://github.com/GoogleCloudPlatform/gcsfuse/pull/2256#discussion_r1703963103 ?

Reply:
As discussed offline, we can't defined cases like below in go, it will be comma separated. Have done the same.
```
    case ic.Folder != nil:
    case (ic.MinObject != nil && ic.FullName.IsDir()):
        in = fs.createExplicitDirInode(id, ic)
```